entrypoint.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh -l
+set -euo pipefail
+
+echo "setting prerequisites."
+export BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr -d '[:punct:]')
+
+
+if [ $BRANCH_NAME != 'master' ];
+then 
+    export VERSION_SUFFIX=$BRANCH_NAME.$GITHUB_RUN_NUMBER
+else
+    export VERSION_SUFFIX=$BRANCH_NAME.$GITHUB_RUN_NUMBER.1
+fi
+
+echo "Print Branch name : $BRANCH_NAME"
+echo "Print VERSION_SUFFIX1: $VERSION_SUFFIX"
+
+echo "Current directory : $GITHUB_WORKSPACE"
+
+echo "Start: code build."
+if [ -z "${CUSTOM_DOCKER_FILE_PATH}" ];
+then
+   echo "Running lazy build steps."
+   cd $GITHUB_WORKSPACE && dotnet build --configuration Release
+   echo "Start run tests."
+   cd $GITHUB_WORKSPACE && dotnet test
+
+else
+   echo "Running custom build steps."
+   chmod +x /lazy-scripts/build.sh && /lazy-scripts/build.sh ${DOCKER_REPO_NAME} ${DOCKER_IMAGE_NAME} sha-${GITHUB_SHA:0:7}
```

Comment:
Docker build should happen after sonar scan

Reply:
isn't it other way. First we build the code to see if there are any compilation errors or packaging errors and  then proceed to run tests.

Reply:
Building code is different from docker build. In docker build, you are building the image.

Reply:
Moved building image after sonar scan.